File: C11/백준/Silver/11403. 경로 찾기/경로 찾기.c
#include <stdio.h>
#include <string.h>

#define MAX 100

int N;
int graph[MAX][MAX];
int result[MAX][MAX];
int visited[MAX];

void dfs(int start, int curr) {
    for (int next = 0; next < N; next++) {
        if (graph[curr][next] == 1 && !visited[next]) {
            visited[next] = 1;
            result[start][next] = 1;
            dfs(start, next);
        }
    }
}

int main() {
    scanf("%d", &N);

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            scanf("%d", &graph[i][j]);
        }
    }

    for (int i = 0; i < N; i++) {
        memset(visited, 0, sizeof(visited));
        dfs(i, i);
    }

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            printf("%d ", result[i][j]);
        }
        printf("\n");
    }

    return 0;
}

File: Java/백준/Silver/11403. 경로 찾기/경로 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {
    static int N;
    static List<List<Integer>> graph = new ArrayList<>();
    static int[][] result;
    static boolean[] visited;

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        N = Integer.parseInt(br.readLine()); //정점의 개수
        result = new int[N][N];

        for (int i = 0; i < N; i++) { //리스트 초기화
            graph.add(new ArrayList<>());
        }

        for (int i = 0; i < N; i++) {
            String[] input = br.readLine().split(" ");
            for (int j = 0; j < N; j++) {
                if (input[j].equals("1")) { //간선 추가
                    graph.get(i).add(j);
                }
            }
        }

        for (int i = 0; i < N; i++) {
            visited = new boolean[N];
            dfs(i, i);
        }

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                System.out.print(result[i][j] + " ");
            }
            System.out.println();
        }
    }

    static void dfs(int start, int curr) {
        for (int next : graph.get(curr)) {
            if (!visited[next]) {
                visited[next] = true;
                result[start][next] = 1;
                dfs(start, next);
            }
        }
    }
}